shared/bundle_analysis/parsers/v3.py

```diff
@@ -0,0 +1,424 @@
+            self.asset.gzip_size = int(value)
+        elif (prefix, event) == ("assets.item", "end_map"):
+            self.asset_list.append(
+                dict(
+                    session_id=self.asset.session_id,
+                    name=self.asset.name,
+                    normalized_name=self.asset.normalized_name,
+                    size=self.asset.size,
+                    gzip_size=self.asset.gzip_size,
+                    uuid=str(uuid.uuid4()),
+                    asset_type=self._asset_type(self.asset.name),
+                )
+            )
+
+            # reset parser state
+            self.asset = None
+
+    def _parse_chunks_event(self, prefix: str, event: str, value: str):
+        if (prefix, event) == ("chunks.item", "start_map"):
+            # new chunk
+            assert self.chunk is None
+            self.chunk = Chunk(session_id=self.session.id)
+        elif prefix == "chunks.item.id":
+            self.chunk.external_id = value
+        elif prefix == "chunks.item.uniqueId":
+            self.chunk.unique_external_id = value
+        elif prefix == "chunks.item.initial":
+            self.chunk.initial = value
+        elif prefix == "chunks.item.entry":
+            self.chunk.entry = value
+        elif prefix == "chunks.item.files.item":
+            self.chunk_asset_names.append(value)
+        elif prefix == "chunks.item.dynamicImports.item":
+            self.dynamic_imports_mapping[self.chunk].append(value)
+        elif (prefix, event) == ("chunks.item", "end_map"):
+            self.chunk_list.append(self.chunk)
+
+            self.chunk_asset_names_index[self.chunk.unique_external_id] = (
+                self.chunk_asset_names
+            )
+            # reset parser state
+            self.chunk = None
+            self.chunk_asset_names = []
+
+    def _parse_modules_event(self, prefix: str, event: str, value: str):
+        if (prefix, event) == ("modules.item", "start_map"):
+            # new module
+            assert self.module is None
+            self.module = Module(session_id=self.session.id)
+        elif prefix == "modules.item.name":
+            self.module.name = value
+        elif prefix == "modules.item.size":
+            self.module.size = int(value)
+        elif prefix == "modules.item.chunkUniqueIds.item":
+            self.module_chunk_unique_external_ids.append(value)
+        elif (prefix, event) == ("modules.item", "end_map"):
+            self.module_list.append(
+                dict(
+                    session_id=self.module.session_id,
+                    name=self.module.name,
+                    size=self.module.size,
+                )
+            )
+
+            self.module_chunk_unique_external_ids_index[self.module.name] = (
+                self.module_chunk_unique_external_ids
+            )
+            # reset parser state
+            self.module = None
+            self.module_chunk_unique_external_ids = []
+
```

Comment:
This function below is a new one, it collects all the pairs of (chunk, asset) that is a dynamic import to be saved into the DB at the end.